3p/frame.max.html

```diff
@@ -2,7 +2,7 @@
 <head>
 <meta charset="utf-8">
 <meta name="robots" content="noindex">
-<script src="./integration.js"></script>
+<script src="./f.js"></script>
```

Comment:
`f.js` has access to `src/config.js`, but it attempts to read from `window.AMP_CONFIG`. Maybe if we recursively merged the properties from `window[.parent[.parent...]].AMP_CONFIG` in the `src/config.js` module, then that could work.

```javascript
/**
 * Tail-recursively travels window->window.parent to mixin
 * parent AMP_CONFIG configurations into children, until we
 * arrive a single result. Preference is given to the youngest
 * children.
 * @param  {Window} win 
 * @return {AMP_CONFIG}
 */
const winConfig = win => {
  const result = {};
  const currConfig = win.AMP_CONFIG || {};

  if (!win.parent) {
    return currConfig;
  }

  const parentConfig = winConfig(win.parent);

  for (const key in parentConfig) {
    result[key] = parentConfig[key];
  }

  for (const key in currConfig) {
    result[key] = currConfig[key];
  }

  return result;
};

const env = winConfig(window);

export const urls = {
  thirdParty: env.thirdPartyUrl || 'https://3p.ampproject.net',
  thirdPartyFrameHost: env.thirdPartyFrameHost || 'ampproject.net',
  thirdPartyFrameRegex: env.thirdPartyFrameRegex ||
                        /^d-\d+\.ampproject\.net$/,
  cdn: env.cdnUrl || 'https://cdn.ampproject.org',
  errorReporting: env.errorReportingUrl ||
                  'https://amp-error-reporting.appspot.com/r',
};
```

This just reaches out to `window.parent` and `window.parent.parent` until we hit a base condition, and merges all of the `AMP_CONFIG`'s. This way, when `f.js` runs, its `src/config.js` module is properly informed with information from its parent iframe.

I might totally wrong here as I don't have a full grasp on the repo, but I think that works!